### PR TITLE
feat(worklist): New worklist buttons and tooltips

### PR DESCRIPTION
**Pull request by IbrahimCSAE**
### Context

Address https://github.com/OHIF/Viewers/issues/3960
Also fixes issue with dialogs and modals not dismissing on mode exit.
### Changes & Results

New button design
New tooltips

Screenshot from 2024-03-14 04-19-47

Screenshot from 2024-03-14 04-19-53

### Note

Investigational use changes is a separate pull later on.

### CODE REVIEW

File: platform/ui/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -66,19 +67,20 @@ const Button = ({
   name,
   className,
   onClick,
+  tooltip = null,
```
Comment (sedghi) [inline platform/ui/src/components/Button/Button.tsx:70]: it's not button tooltip right? it is icon tooltip

File: platform/app/src/routes/WorkList/WorkList.tsx [hunk 2]
```diff
@@ -382,16 +386,30 @@ function WorkList({
...
-                      variant={isValidMode ? 'contained' : 'disabled'}
+                    <Button
+                      type={ButtonEnums.type.primary}
+                      size={ButtonEnums.size.medium}
                       disabled={!isValidMode}
-                      endIcon={<Icon name="launch-arrow" />} // launch-arrow | launch-info
+                      tooltip={
+                        !isValidMode ? (
+                          <div className="font-inter flex w-[206px] whitespace-normal text-left text-xs font-normal text-white	">
+                            This study does not contain modalities supported by {mode.displayName}{' '}
```
Comment (sedghi) [inline platform/app/src/routes/WorkList/WorkList.tsx:396]: I think the mode itself should provide this text, maybe isValid on the mode should return an object a boolean and a description.

Comment (sedghi) [reply platform/app/src/routes/WorkList/WorkList.tsx:396]: For intstance for tmtv, we can say it is activated on studies with both ct and pt

For SEG it is a little bit more 

- on SM -> no support for seg on SM yet
- on DX, MG -> no support for non-volumetric segmentation yet

etc etc

Check wordings with Dan

**Review by sedghi — COMMENTED** [review]
Looks good, see my comments please

**Comment by sedghi** [comment]
@IbrahimCSAE Can you please add a configuration inside the app config that can people choose in their configurations like default.js and the configuration is about whether to group the enabled modes together at the front and then show the disabled ones. I think that adds a lot of value.

**Comment by IbrahimCSAE** [comment]
> @IbrahimCSAE Can you please add a configuration inside the app config that can people choose in their configurations like default.js and the configuration is about whether to group the enabled modes together at the front and then show the disabled ones. I think that adds a lot of value.

Done

File: platform/app/src/routes/WorkList/WorkList.tsx (hunk 2 above)
Comment (IbrahimCSAE) [reply platform/app/src/routes/WorkList/WorkList.tsx]: Done, wording is my own though, I didn't check with Dan on this yet.

File: platform/ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (IbrahimCSAE) [reply platform/ui/src/components/Button/Button.tsx]: Fixed

**Review by sedghi — APPROVED** [review]
Looks great thanks

**Pull request merged**